src/cr/cube/crunch_cube.py

```diff
@@ -176,92 +198,131 @@ def _get_values(self, weighted, margin=False):
-                if not transform or dim.type == 'categorical_array':
-                    continue
-
-                # Perform transformations
-                ind_offset = 0
-                for indices in dim.hs_indices:
-                    ind_subtotal_elements = np.array(indices['inds'])
-                    ind_insertion = indices['anchor_ind'] + ind_offset
-                    ind_subtotal_elements = (np.array(ind_subtotal_elements) +
-                                             ind_offset)
-                    if i == 0:
-                        value = sum(res[ind_subtotal_elements])
-                        res = np.insert(res, ind_insertion + 1, value, axis=i)
-                    else:
-                        value = np.sum(res[:, ind_subtotal_elements], axis=1)
-                        res = np.insert(res, ind_insertion + 1, value, axis=i)
-                    valid_indices = self._fix_valid_indices(valid_indices,
-                                                            ind_insertion, i)
-                    ind_offset += 1
-
-        res = res[np.ix_(*valid_indices)]
-
-        adjustment = 1 if adjusted else 0
-        res = self._fix_shape(res) + adjustment
+        res = self._transform(res, include_transforms_for_dims, valid_indices)
+        res = self._fix_shape(res) + (1 if adjusted else 0)
 
         if prune and not self.has_mr:
-            # Remove columns and rows where marginal value is 0 or np.nan
-            if len(self.dimensions) == 1 or len(res.shape) == 1:
-                margin = self.margin(
-                    include_transforms_for_dims=include_transforms_for_dims,
-                    axis=1,
-                )
-                ind_prune = np.logical_or(margin == 0, np.isnan(margin))
-                return res[~ind_prune]
-
-            col_margin = self.margin(
-                include_transforms_for_dims=include_transforms_for_dims,
-                axis=0,
-            )
-            ind_prune = np.logical_or(col_margin == 0, np.isnan(col_margin))
-            res = res[:, ~ind_prune]
-            row_margin = self.margin(
-                include_transforms_for_dims=include_transforms_for_dims,
-                axis=1
-            )
-            ind_prune = np.logical_or(row_margin == 0, np.isnan(row_margin))
-            return res[~ind_prune, :]
+            return self._prune(res, include_transforms_for_dims)
 
         return res
 
+    def _prune(self, res, transforms=None):
+        '''Prune the result based on margins content.
+
+        Pruning is the removal of rows or columns, whose corresponding
+        marginal elements are either 0 or not defined (np.nan).
+        '''
+
+        if len(self.dimensions) == 1 or len(res.shape) == 1:
+            # For 1D, margin is calculated as the row margin.
+            margin = self.margin(include_transforms_for_dims=transforms, axis=1)
+            ind_prune = np.logical_or(margin == 0, np.isnan(margin))
```

Comment:
why not just do "col_margin == 0 or  np.isnan(col_margin)"  not sure why logical or is needed here

Reply:
Because it doesn't work with `numpy` (or I haven't figured it out). I mean, it works for single value, but not "bitwise". Will try to find an option to make it work...